### ISSUE TITLE
Bug: SQL error in Projects area

### ISSUE THREAD

**Issue by marcosgdf** [Dolibarr/dolibarr#5087]
Version: 3.9
DB Type: Postgres

```
( ! ) Xdebug: user triggered in /var/www/dolibarr/htdocs/core/lib/functions.lib.php on line 2746
Call Stack
#   Time    Memory  Function    Location
1   0.0003  241152  {main}( )   .../index.php:0
2   0.1510  1531384 print_projecttasks_array( $db = class DoliDBPgsql { public $type = 'pgsql'; public $forcecharset = 'utf8'; public $forcecollate = 'utf8_general_ci'; private $_results = FALSE; public $unescapeslashquot = NULL; public $standard_conforming_strings = NULL; public $db = resource(1) of type (pgsql link); public $connected = TRUE; public $database_selected = TRUE; public $database_name = 'dolibarr39'; public $database_user = 'postgres'; public $database_host = 'localhost'; public $database_port = '5432'; public $transaction_opened = 0; public $lastquery = 'SELECT p.rowid as projectid, p.ref, p.title, p.fk_soc, s.nom as socname, p.fk_user_creat, p.public, p.fk_statut as status, p.fk_opp_status as opp_status, p.opp_amount, COUNT(t.rowid) as nb, SUM(t.planned_workload) as planned_workload, SUM(t.planned_workload * t.progress / 100) as declared_progess_workload FROM llx_projet as p LEFT JOIN llx_societe as s ON s.rowid = p.fk_soc LEFT JOIN llx_projet_task as t ON p.rowid = t.fk_projet WHERE p.rowid IN (-1) GROUP BY p.rowid, p.ref, p.title, p.fk_soc, p.fk_user_cre...'; public $lastqueryerror = 'SELECT p.rowid as projectid, p.ref, p.title, p.fk_soc, s.nom as socname, p.fk_user_creat, p.public, p.fk_statut as status, p.fk_opp_status as opp_status, p.opp_amount, COUNT(t.rowid) as nb, SUM(t.planned_workload) as planned_workload, SUM(t.planned_workload * t.progress / 100) as declared_progess_workload FROM llx_projet as p LEFT JOIN llx_societe as s ON s.rowid = p.fk_soc LEFT JOIN llx_projet_task as t ON p.rowid = t.fk_projet WHERE p.rowid IN (-1) GROUP BY p.rowid, p.ref, p.title, p.fk_soc, p.fk_user_cre...'; public $lasterror = 'ERROR: 42803: column "s.nom" must appear in the GROUP BY clause or be used in an aggregate function\nLINE 1: ...T p.rowid as projectid, p.ref, p.title, p.fk_soc, s.nom as s...\n ^\nLOCATION: check_ungrouped_columns_walker, parse_agg.c:911'; public $lasterrno = 'DB_ERROR_42803'; public $ok = TRUE; public $error = NULL; private ${DoliDB}:_results = NULL }, $form = class Form { public $db = class DoliDBPgsql { public $type = 'pgsql'; public $forcecharset = 'utf8'; public $forcecollate = 'utf8_general_ci'; private $_results = FALSE; public $unescapeslashquot = NULL; public $standard_conforming_strings = NULL; public $db = resource(1) of type (pgsql link); public $connected = TRUE; public $database_selected = TRUE; public $database_name = 'dolibarr39'; public $database_user = 'postgres'; public $database_host = 'localhost'; public $database_port = '5432'; public $transaction_opened = 0; public $lastquery = 'SELECT p.rowid as projectid, p.ref, p.title, p.fk_soc, s.nom as socname, p.fk_user_creat, p.public, p.fk_statut as status, p.fk_opp_status as opp_status, p.opp_amount, COUNT(t.rowid) as nb, SUM(t.planned_workload) as planned_workload, SUM(t.planned_workload * t.progress / 100) as declared_progess_workload FROM llx_projet as p LEFT JOIN llx_societe as s ON s.rowid = p.fk_soc LEFT JOIN llx_projet_task as t ON p.rowid = t.fk_projet WHERE p.rowid IN (-1) GROUP BY p.rowid, p.ref, p.title, p.fk_soc, p.fk_user_cre...'; public $lastqueryerror = 'SELECT p.rowid as projectid, p.ref, p.title, p.fk_soc, s.nom as socname, p.fk_user_creat, p.public, p.fk_statut as status, p.fk_opp_status as opp_status, p.opp_amount, COUNT(t.rowid) as nb, SUM(t.planned_workload) as planned_workload, SUM(t.planned_workload * t.progress / 100) as declared_progess_workload FROM llx_projet as p LEFT JOIN llx_societe as s ON s.rowid = p.fk_soc LEFT JOIN llx_projet_task as t ON p.rowid = t.fk_projet WHERE p.rowid IN (-1) GROUP BY p.rowid, p.ref, p.title, p.fk_soc, p.fk_user_cre...'; public $lasterror = 'ERROR: 42803: column "s.nom" must appear in the GROUP BY clause or be used in an aggregate function\nLINE 1: ...T p.rowid as projectid, p.ref, p.title, p.fk_soc, s.nom as s...\n ^\nLOCATION: check_ungrouped_columns_walker, parse_agg.c:911'; public $lasterrno = 'DB_ERROR_42803'; public $ok = TRUE; public $error = NULL; private ${DoliDB}:_results = NULL }; public $error = NULL; public $num = NULL; public $cache_types_paiements = array (); public $cache_conditions_paiements = array (); public $cache_availability = array (); public $cache_demand_reason = array (); public $cache_types_fees = array (); public $cache_vatrates = array () }, $socid = 0, $projectsListId = '0', $mytasks = 0, $statut = 0, $listofoppstatus = array (1 => '0.00', 2 => '20.00', 3 => '40.00', 4 => '60.00', 5 => '50.00', 6 => '100.00', 7 => '0.00'), $hiddenfields = array (0 => 'projectlabel', 1 => 'plannedworkload', 2 => 'declaredprogress') )    .../index.php:135
3   0.1551  1539760 dol_print_error( $db = class DoliDBPgsql { public $type = 'pgsql'; public $forcecharset = 'utf8'; public $forcecollate = 'utf8_general_ci'; private $_results = FALSE; public $unescapeslashquot = NULL; public $standard_conforming_strings = NULL; public $db = resource(1) of type (pgsql link); public $connected = TRUE; public $database_selected = TRUE; public $database_name = 'dolibarr39'; public $database_user = 'postgres'; public $database_host = 'localhost'; public $database_port = '5432'; public $transaction_opened = 0; public $lastquery = 'SELECT p.rowid as projectid, p.ref, p.title, p.fk_soc, s.nom as socname, p.fk_user_creat, p.public, p.fk_statut as status, p.fk_opp_status as opp_status, p.opp_amount, COUNT(t.rowid) as nb, SUM(t.planned_workload) as planned_workload, SUM(t.planned_workload * t.progress / 100) as declared_progess_workload FROM llx_projet as p LEFT JOIN llx_societe as s ON s.rowid = p.fk_soc LEFT JOIN llx_projet_task as t ON p.rowid = t.fk_projet WHERE p.rowid IN (-1) GROUP BY p.rowid, p.ref, p.title, p.fk_soc, p.fk_user_cre...'; public $lastqueryerror = 'SELECT p.rowid as projectid, p.ref, p.title, p.fk_soc, s.nom as socname, p.fk_user_creat, p.public, p.fk_statut as status, p.fk_opp_status as opp_status, p.opp_amount, COUNT(t.rowid) as nb, SUM(t.planned_workload) as planned_workload, SUM(t.planned_workload * t.progress / 100) as declared_progess_workload FROM llx_projet as p LEFT JOIN llx_societe as s ON s.rowid = p.fk_soc LEFT JOIN llx_projet_task as t ON p.rowid = t.fk_projet WHERE p.rowid IN (-1) GROUP BY p.rowid, p.ref, p.title, p.fk_soc, p.fk_user_cre...'; public $lasterror = 'ERROR: 42803: column "s.nom" must appear in the GROUP BY clause or be used in an aggregate function\nLINE 1: ...T p.rowid as projectid, p.ref, p.title, p.fk_soc, s.nom as s...\n ^\nLOCATION: check_ungrouped_columns_walker, parse_agg.c:911'; public $lasterrno = 'DB_ERROR_42803'; public $ok = TRUE; public $error = NULL; private ${DoliDB}:_results = NULL }, $error = ???, $errors = ??? )   .../project.lib.php:1095
4   0.1655  1594408 xdebug_print_function_stack ( ) .../functions.lib.php:2746
Dolibarr ha detectado un error técnico.
He aquí la información que podrá ayudar al diagnóstico:
Fecha: 20160423142729
Dolibarr: 3.9.1
Nivel de funciones: 0
PHP: 5.6.20-3+donate.sury.org~trusty+1
Server: Apache/2.4.20 (Ubuntu)
OS: Linux dolibarr 3.19.0-25-generic #26~14.04.1-Ubuntu SMP Fri Jul 24 21:16:20 UTC 2015 x86_64
UserAgent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/49.0.2623.112 Safari/537.36

URL solicitada: /projet/index.php?mainmenu=project&leftmenu=
Referer: http://dolibarr.local/admin/modules.php?mode=common&page_y=660
Gestor de menú: eldy_menu.php

Tipo de gestor de base de datos: pgsql
Petición último acceso a la base de datos erróneo: SELECT p.rowid as projectid, p.ref, p.title, p.fk_soc, s.nom as socname, p.fk_user_creat, p.public, p.fk_statut as status, p.fk_opp_status as opp_status, p.opp_amount, COUNT(t.rowid) as nb, SUM(t.planned_workload) as planned_workload, SUM(t.planned_workload * t.progress / 100) as declared_progess_workload FROM llx_projet as p LEFT JOIN llx_societe as s ON s.rowid = p.fk_soc LEFT JOIN llx_projet_task as t ON p.rowid = t.fk_projet WHERE p.rowid IN (-1) GROUP BY p.rowid, p.ref, p.title, p.fk_soc, p.fk_user_creat, p.public, p.fk_statut, p.fk_opp_status, p.opp_amount ORDER BY p.title, p.ref
Código devuelto último acceso a la base de datos erróneo: DB_ERROR_42803
Información sobre el último acceso a la base de datos erróneo: ERROR: 42803: column "s.nom" must appear in the GROUP BY clause or be used in an aggregate function LINE 1: ...T p.rowid as projectid, p.ref, p.title, p.fk_soc, s.nom as s... ^ LOCATION: check_ungrouped_columns_walker, parse_agg.c:911

XDebug informations:
File: /var/www/dolibarr/htdocs/core/lib/project.lib.php
Line: 1095
Function: print_projecttasks_array
```

**Comment by eldy** [comment]
Looks to be fixed in 3.9.2

**Comment by marcosgdf** [comment]
Still happening in develop version
